crates/router/src/utils.rs

```diff
@@ -138,6 +138,19 @@ pub fn to_currency_base_unit(
     Ok(format!("{amount_f64:.2}"))
 }
 
+/// Convert the amount to its base denomination based on Currency and check for zero decimal currency and return String
+pub fn to_currency_base_unit_with_zero_decimal_check(
+    amount: i64,
+    currency: diesel_models::enums::Currency,
+) -> Result<String, error_stack::Report<errors::ValidationError>> {
+    let amount_f64 = to_currency_base_unit_asf64(amount, currency)?;
+    if currency.is_zero_decimal_currency() {
+        Ok(amount_f64.to_string())
+    } else {
+        Ok(format!("{amount_f64:.2}"))
```

Comment:
Should you vary based on 2 or 3 decimal currency?

Reply:
As far as we checked there are no connectors which accepts 3 digit decimals. Currently this method is being used in PayPal only, which don't accept 3 digit decimals for 3 decimal currency like JPY.